<!-- Runs obstacle detector and tracker with a static scan publisher. -->
<launch>
  <node pkg="obstacle_detector" type="static_scan_publisher" name="static_scan_publisher" output="screen">
    <remap from="scan" to="front_scan"/>
  </node>
  
  <node pkg="tf" type="static_transform_publisher" name="world_to_scanner" output="screen" args="0 0 0 0 0 0 world scanner 100"/>
  
  <node pkg="obstacle_detector" type="scans_merger" name="scans_merger" output="screen">
    <param name="frame_id"          value="scanner"/>

    <param name="publish_scan"      value="true"/>
    <param name="publish_pcl"       value="false"/>

    <param name="ranges_num"        value="1000"/>

    <param name="max_scanner_range" value="10.0"/>
    <param name="min_scanner_range" value="0.03"/>

    <param name="max_x_range"       value="10.0"/>
    <param name="min_x_range"       value="-10.0"/>
    <param name="max_y_range"       value="10.0"/>
    <param name="min_y_range"       value="-10.0"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_detector" name="obstacle_detector" output="screen">
    <param name="frame_id"             value="world"/>

    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>
    <param name="use_split_and_merge"  value="false"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"      value="false"/>

    <param name="min_group_points"     value="5"/>
    <param name="max_group_distance"   value="0.160"/>    
    <param name="distance_proportion"  value="0.006136"/>
    <param name="max_split_distance"   value="0.070"/>

    <param name="max_merge_separation" value="0.150"/>
    <param name="max_merge_spread"     value="0.070"/>
    <param name="max_circle_radius"    value="0.300"/>
    <param name="radius_enlargement"   value="0.030"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_visualizer" name="raw_obstacle_visualizer" output="screen">
    <param name="tracked_circles_color"   value="7"/>
    <param name="untracked_circles_color" value="7"/>
    <param name="segments_color"          value="6"/>
    <param name="alpha"                   value="0.7"/>
    <param name="z_layer"                 value="0.5"/>
  </node>
<!-- -->
</launch>

